### PR TITLE
possible portal fixes

### LINKED ISSUE #3486: Portal ACL broken 5.0.3 and 5.0.2
- Rest URI such as patient/10 refers to Id and not pid.
https://github.com/openemr/openemr/blob/78bf091db7aa3e4e7dbc9cc1f3897874132e0e70/portal/patient/fwk/libs/verysimple/Phreeze/GenericRouter.php#L132
- api/onsiteportalactivities is how templates are stored/retrieved needs p_all
https://github.com/openemr/openemr/blob/78bf091db7aa3e4e7dbc9cc1f3897874132e0e70/portal/patient/fwk/libs/verysimple/Phreeze/GenericRouter.php#L97

### PR DESCRIPTION
**Pull request by bradymiller**
Fixes #3486

@sjpadgett , I think this gets us what we need and maintains protection from calling other endpoints within the portal context. Testing well (able to see and bring in the template doc from portal into the emr), although I admittedly don't know the workflow very well on this.

### CODE REVIEW

**Comment by bradymiller** [comment]
Note that p_limited is only utilized on the endpoints where a patient id is
used in the uri. This is why the other routes were secured in the
downstream controllers and use p_all instead of p_limited. Check out the
that routes script for details (sorry, can't link to anything since no
github access here). Note this is essentially a public facing api when
registration is turned on, so having 2 layers of defense would be a good
thing.

**Comment by sjpadgett** [comment]
> Note that p_limited is only utilized on the endpoints where a patient id is used in the uri.

Yes but a uri id for patient record will be a different id than for portalpatient which is a different record set for profile audits(review records) actually id is not needed for these. Same for onsite documents.
Remember all these stem from dataset models and collections of models. This allow us to clone a model, modify and save as a separate record for history and review(part of audit trail). Until review is saved to chart we can have two records with different ids/locations.
Bottom line, trying to carry around a patient id will not work with all routes. This is even more true with onsite documents.

This is the only hiccup I see for this to work. The audit records should be based on pid and audit state and not the audit id. I need to change the routes and workflow for these audits to be more consistent with normal endpoint routes.

I'll let you know but, let me know how you want changes. I can just email them or push to your branch.

**Comment by bradymiller** [comment]
Just push them to my branch, if possible. This will likely all become more
clear to me after I see the code changes :)

**Comment by sjpadgett** [comment]
> clear to me after I see the code changes

When it does, please clue me in too!

**Comment by bradymiller** [comment]
the blind leading the blind :)

**Comment by sjpadgett** [comment]
@bradymiller I've tested on 5.0.2. Found a bug in patientdata.js with id. Cleaned up some routes and controllers.
Once in for 5.0.3 i'll test again then use my PR in 5.0.2 for backport.(already have written for 502)

I really like working with you Brady but, can we agree not to do these anymore and do something fun!:)

**Comment by bradymiller** [comment]
hi @sjpadgett , Looking nice! I think the only thing left to do is to bootstrap the downstream OnsitePortalActivity.Read function here https://github.com/openemr/openemr/blob/55040259de483e28cf513ac2623590991050ed13/portal/patient/libs/Controller/OnsitePortalActivityController.php#L113-L115

Guessing need to add something like this there:
https://github.com/openemr/openemr/blob/55040259de483e28cf513ac2623590991050ed13/portal/patient/libs/Controller/OnsitePortalActivityController.php#L190-L195

I can take a stab at it if needed. Otherwise I think we are good to go if testing well for you.

**Comment by bradymiller** [comment]
btw, regarding fun activities besides this of course:
tenor (16)

Planning to try to place patient support in the main API :)
https://community.open-emr.org/t/project-hybrid-app-with-image-processing/13758/77

**Comment by sjpadgett** [comment]
> Guessing need to add something like this there:

Could have swore I added that. Suppose it doesn't do any good on my dev machine!
I'll fix travis(grrr) and this then bring in after RTL is brought in to solve for any conflict.

**Comment by sjpadgett** [comment]
thanks brady!
